src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts

```diff
@@ -0,0 +1,269 @@
+        this.exportService.toExcel<SuperTask>(
+          'hashtopolis-supertasks',
+          this.tableColumns,
+          event.data,
+          SupertasksTableColumnLabel
+        );
+        break;
+      case ExportMenuAction.CSV:
+        this.exportService.toCsv<SuperTask>(
+          'hashtopolis-supertasks',
+          this.tableColumns,
+          event.data,
+          SupertasksTableColumnLabel
+        );
+        break;
+      case ExportMenuAction.COPY:
+        this.exportService
+          .toClipboard<SuperTask>(
+            this.tableColumns,
+            event.data,
+            SupertasksTableColumnLabel
+          )
+          .then(() => {
+            this.snackBar.open(
+              'The selected rows are copied to the clipboard',
+              'Close'
+            );
+          });
+        break;
+    }
+  }
+
+  rowActionClicked(event: ActionMenuEvent<SuperTask>): void {
+    switch (event.menuItem.action) {
+      case RowActionMenuAction.EDIT:
+        this.rowActionEdit(event.data);
+        break;
+      case RowActionMenuAction.APPLY_TO_HASHLIST:
+        this.rowActionApplyToHashlist(event.data);
+        break;
+      case RowActionMenuAction.EDIT_SUBTASKS:
+        this.rowActionEditSubtasks(event.data);
+        break;
+      case RowActionMenuAction.DELETE:
+        this.openDialog({
+          rows: [event.data],
+          title: `Deleting Supertask ${event.data.supertaskName} ...`,
+          icon: 'warning',
+          body: `Are you sure you want to delete it? Note that this action cannot be undone.`,
+          warn: true,
+          action: event.menuItem.action
+        });
+        break;
+    }
+  }
+
+  bulkActionClicked(event: ActionMenuEvent<SuperTask[]>): void {
+    switch (event.menuItem.action) {
+      case BulkActionMenuAction.DELETE:
+        this.openDialog({
+          rows: event.data,
+          title: `Deleting ${event.data.length} supertasks ...`,
+          icon: 'warning',
+          body: `Are you sure you want to delete the above supertasks? Note that this action cannot be undone.`,
+          warn: true,
+          listAttribute: 'supertaskName',
+          action: event.menuItem.action
+        });
+        break;
+    }
+  }
+
+  /**
+   * @todo Implement error handling.
+   */
+  private bulkActionDelete(supertask: SuperTask[]): void {
+    const requests = supertask.map((supertask: SuperTask) => {
+      return this.gs.delete(SERV.SUPER_TASKS, supertask._id);
+    });
+
+    this.subscriptions.push(
+      forkJoin(requests)
+        .pipe(
+          catchError((error) => {
+            console.error('Error during deletion:', error);
+            return [];
+          })
+        )
+        .subscribe((results) => {
+          this.snackBar.open(
+            `Successfully deleted ${results.length} supertasks!`,
+            'Close'
+          );
+          this.reload();
+        })
+    );
+  }
+
+  @Cacheable(['_id'])
+  async renderSupertaskLink(
```

Comment:
Maybe this one can be declared in the base class with the other links?

Reply:
Yes, makes sense, completely missed that. thanks